src/Domain/CampaignRepository.php

```diff
@@ -361,9 +361,11 @@ static function (?string $string) {
             $postalAddress
         );
 
+        // For now, treat whole address as null if there's no `line1`. This can happen with allowed
+        // Salesforce addresses for now. For example, a charity may fill in just a country in the
+        // portal and save that as their own address. We're better off omitting it from donor emails
+        // if that happens.
```

Comment:
It's a bit making me wish we'd saved the address as one string or an list of strings instead of named fields. I don't really like ignoring data if our UI accepted it without complaint. Wondering if we should make this a validation rule on the charity object. Or we just allow line1 to be null in Matchbot. Any falsy fields already get filtered out when the address is displayed. 

We shouldn't assume this is just for donor emails, that's the current use but there's nothing to say that in the code nearby, it could equally be for display via frontend.

Reply:
@NoelLH what do you think about keeping the assert but catching the exception and making it only generate an alarm for now? Then if we get too many of those alarms we can put a validation rule on the campaign in SF?

Reply:
Ah yep that is probably better, good idea

Reply:
Done in 4371bee65e2c6510e39c0961c15f14629a1b8595